projects/aca-shared/src/lib/services/app.service.ts

```diff
@@ -166,9 +166,11 @@ export class AppService implements OnDestroy {
       if (isReady) {
         this.loadRepositoryStatus();
         this.loadUserProfile();
-        setTimeout(() => {
-          this.openMobileAppDialog();
-        });
+        if (document.readyState === 'complete') {
+          setTimeout(() => {
+            this.openMobileAppDialog();
+          }, 10000);
```

Comment:
Do we really need 10 seconds of delay? We should avoid high delays and check if there is better way. Why you need this? Is that required because you need wait until page is loaded? If yes then maybe you can use following:

https://stackoverflow.com/questions/50345082/angular-4-check-if-all-page-resources-are-fully-loaded-make-loader-spins-unti

(ngAfterViewInit)

@MichalKinas what do you think?

Reply:
Yeah I agree, timeout with 10 seconds doesn't seem right and there is a high chance that `ngAfterViewInit` will help